website/blog/buck2.md

```diff
@@ -0,0 +1,564 @@
+            = more info about errors; 3 = more info about everything [default: 1]
+```
+
+The build progress library was custom-made for Buck2 and could sort and color-code slower actions.
+Here is a quick sped up demo:
+
+![](../static/img/blog/buck2-success.gif)
+
+#### First-class Remote Execution
+
+Buck2 was not kidding about it being a "Remote Execution first" build tool.
+
+Thanks to this assumption, Buck2 was able to cut corners by not providing any local-sandboxing implementations by default.
+What seems to be an unwanted side-effect at scale, actually made the tool more friendly to new learners.
+The lack of local sandboxing allows Buck2 to behave locally closer to other task-runner build tools such as Make or Just.
+
+However, any mistakes that are made when authoring build definitions locally will very soon be revealed once moved to Remote Build Execution.
+Buck2 follows the same industry standard [Remote Execution API](https://github.com/bazelbuild/remote-apis) that was adopted by the like of Bazel, Goma, Pants, Please, Recc...
+This means that Buck2 is compatible with BuildBuddy Remote Cache and Remote Build Execution offering! 🎉
+
+So let's test it out!
+
+First, let's establish a setup to benchmark our test
+
+```python
+SIZE_RANGE = range(0, 1000)
+
+XYZ_SHA256 = "72036ae48c55d632cc333cac04c78be336b0c76f39b225d023e218efe077ab1b"
+
+# Add some execution time to each action to simulate a real build.
+SLEEP_CMD = "sleep $$((RANDOM % 5));"
+
+# Stage 1
+[
+    genrule(
+        name = "x_{}".format(n),
+        outs = ["x_{}.txt".format(n)],
+        cmd = SLEEP_CMD + "echo x_{} > $@".format(n),
+    )
+    for n in SIZE_RANGE
+]
+
+[
+    genrule(
+        name = "y_{}".format(n),
+        outs = ["y_{}.txt".format(n)],
+        cmd = SLEEP_CMD + "echo y_{} > $@".format(n),
+    )
+    for n in SIZE_RANGE
+]
+
+# Stage 2
+[
+    genrule(
+        name = "z_{}".format(n),
+        outs = ["z_{}.txt".format(n)],
+        cmd = SLEEP_CMD + "cat $(location :x_{n}) $(location :y_{n}) > $@".format(n = n),
+        tools = [
+            ":x_{}".format(n),
+            ":y_{}".format(n),
+        ],
+    )
+    for n in SIZE_RANGE
+]
+
+# Stage 3
+genrule(
+    name = "xyz",
+    outs = ["xyz.txt"],
+    # cat $(location :z_0) $(location :z_1) ... $(location :z_999) > $@,
+    cmd = SLEEP_CMD + "cat " + " ".join([
+        "$(location :z_{})".format(n)
+        for n in SIZE_RANGE
+    ]) + " > $@",
+    tools = [
+        ":z_{}".format(n)
+        for n in SIZE_RANGE
+    ],
+)
+
+# Stage 4
+sh_test(
+    name = "test",
+    srcs = ["test.sh"],
+    args = [
+        "$(location :xyz)",
+        XYZ_SHA256,
+    ],
+    data = [
+        ":xyz",
+    ],
+)
+```
+
+Our setup is a build graph composed of 4 stages:
+
+- Stage 1: Create 1000 "x" targets and 1000 "y" targets (2000 build targets)
+
+- Stage 2: Create 1000 "z" targets by concat "x" and "y" from Stage 1 (1000 build targets)
```

Comment:
```suggestion
- Stage 2: Create 1000 "z" targets bthaty concat "x" and "y" from Stage 1 (1000 build targets)
```

Reply:
Im not sure about this suggestion. Is this a typo for `s/by/that/`?
```suggestion
- Stage 2: Create 1000 "z" targets that concat "x" and "y" from Stage 1 (1000 build targets)
```